### PR TITLE
[c, cpp]: support VERSION file for CMake ABI versioning

### LINKED ISSUE #320: Please version the ABI for C shared libraries
### 🤔 What's the problem you're trying to solve?

On Linux, system-wide shared libraries are expected to be versioned via `SONAME`.

### ✨ What's your proposed solution?

Set [`SOVERSION`](https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html) in `c/CMakeLists.txt`; this can be as simple as:

```cmake
set_target_properties(gherkin PROPERTIES SOVERSION 1)
```

Commit to incrementing the value of `SOVERSION` each time there is an ABI-incompatible change.

### ⛏ Have you considered any alternatives or workarounds?

[Downstream .so versioning](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_downstream_so_name_versioning) is an option.

### 📚 Any additional context?

It looks like the C++ shared library already does this correctly:

https://github.com/cucumber/gherkin/blob/f99769befdec0c3496027a2d09b716e6ab0c2f45/cpp/src/lib/gherkin/CMakeLists.txt#L34-L42

### PR DESCRIPTION
**Pull request by chybz**
### 🤔 What's changed?

- [c] slight update to existing CMakeFiles.txt to propagate VERSION.  Closes: #320 
- [cpp] add generic support for ABI versioning with VERSION

### ⚡️ What's your motivation? 

make sure ABI is versioned for shared libraries

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)
- :boom: Breaking change (incompatible changes to the API)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://github.com/cucumber/.github/tree/main?tab=coc-ov-file)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: cpp/gherkin-cpp-rule-type.razor [hunk 1]
```diff
@@ -22,7 +22,7 @@ public static string NameOf(Rule rule)
 #include <ostream>
 #include <string_view>
 
-namespace gherkin {
+namespace cucumber::gherkin {
```
Comment (mpkorstanje) [inline cpp/gherkin-cpp-rule-type.razor:25]: Is this a breaking change?

**Comment by chybz** [comment]
@mpkorstanje I'm going to merge. Can you please activate the `VERSION` automatic update for `gherkin/cpp` and `messages/cpp` ?

**Comment by mpkorstanje** [comment]
It's already merged. https://github.com/cucumber/polyglot-release/pull/104

Question though, is the change in namespace that you included breaking in a Semver context? Doesn't matter for merging but does matter when it comes to deciding the next version number.

**Comment by chybz** [comment]
TLDR: yes. Sincerely hope that's not overly problematic.

It is from a pure compilation standpoint, but function signatures are exactly the same as before, so this should be a minimal hassle for the (no more than 2 people I think) current cpp user-base.

**Comment by mpkorstanje** [comment]
Okay cheers! I didn't see your edit until now.

**Comment by mpkorstanje** [comment]
And no problem!
